### PR TITLE
Remove powermock usage from unit tests

### PR DESCRIPTION
**Pull request by KD23243**
### Proposed changes in this pull request

- Remove the usage of the powermock library from the test cases.
- Powermock library is becoming dead and it is so far has not supported Java 21. With our effort to provide support Identity Server product compiling with Java 21, this becomes a blocker.

### CODE REVIEW

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java [hunk 1]
```diff
@@ -254,61 +274,66 @@ public void testIsGroupExisting() throws Exception {
...
-        mockStatic(IdentityDatabaseUtil.class);
-        mockStatic(SCIMCommonUtils.class);
 
         when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
-        whenNew(GroupDAO.class).withNoArguments().thenReturn(mockedGroupDAO);
-        when(mockedGroupDAO.isExistingGroup("NON_EXISTENT_ROLE_NAME", 1)).thenReturn(false);
 
-        SCIMGroupHandler scimGroupHandler = new SCIMGroupHandler(1);
-        scimGroupHandler.updateRoleName("NON_EXISTENT_ROLE_NAME", "NEW_ROLE_NAME");
-        // This method is to test the throwing of an IdentitySCIMException, hence no assertion.
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java:304]: Shall we add this back?

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/IdentitySCIMManagerTest.java [hunk 2]
```diff
@@ -19,44 +19,43 @@
...
+import static org.mockito.ArgumentMatchers.anyInt;
 
 /**
  * Contains the unit test cases for IdentitySCIMManager.
  */
-@PrepareForTest({SCIMCommonUtils.class, PrivilegedCarbonContext.class, SCIMCommonComponentHolder.class,CharonConfiguration.class})
-@PowerMockIgnore({"javax.xml.*","org.w3c.dom.*","org.xml.sax.*"})
-public class IdentitySCIMManagerTest extends PowerMockTestCase {
+@Listeners(MockitoTestNGListener.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/IdentitySCIMManagerTest.java:57]: Is there a specific reason to make the tests lenient?

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java [hunk 3]
```diff
@@ -129,13 +132,20 @@ public void setUpClass() {
 
     @BeforeMethod
     public void setUpMethod() {
-
-        mockStatic(SCIMCommonUtils.class);
-        mockStatic(OrganizationManagementUtil.class);
+        MockitoAnnotations.openMocks(this);
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java:135]: Better to go for `@Listeners(MockitoTestNGListener.class)`

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/utils/AdminAttributeUtilTestForGroup.java [hunk 4]
```diff
@@ -38,18 +39,13 @@
...
 import org.wso2.carbon.user.core.util.UserCoreUtil;
 
-import static org.mockito.Matchers.anyInt;
-import static org.mockito.Matchers.anyObject;
-import static org.mockito.Matchers.anyString;
-import static org.mockito.Mockito.verify;
-import static org.powermock.api.mockito.PowerMockito.mockStatic;
-import static org.powermock.api.mockito.PowerMockito.when;
-import static org.powermock.api.mockito.PowerMockito.whenNew;
+import static org.mockito.Mockito.*;
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/utils/AdminAttributeUtilTestForGroup.java:42]: No `*` imports

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/utils/AdminAttributeUtilTest.java [hunk 5]
```diff
@@ -20,33 +20,25 @@
...
 
 import java.util.Map;
 
-import static org.mockito.Matchers.anyInt;
-import static org.mockito.Matchers.anyString;
-import static org.mockito.Matchers.eq;
-import static org.mockito.Mockito.verify;
-import static org.powermock.api.mockito.PowerMockito.mockStatic;
-import static org.powermock.api.mockito.PowerMockito.when;
+import static org.mockito.Mockito.*;
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/utils/AdminAttributeUtilTest.java:38]: No `*` imports

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java [hunk 6]
```diff
@@ -73,16 +72,32 @@ public class SCIMGroupHandlerTest extends PowerMockTestCase {
...
+
     @BeforeMethod
-    public void setUp() throws Exception {
+    public void setUp() {
+
         initMocks(this);
+        scimCommonUtils = mockStatic(SCIMCommonUtils.class);
+        identityDatabaseUtil = mockStatic(IdentityDatabaseUtil.class);
+        identityTenantUtil = mockStatic(IdentityTenantUtil.class);
+        stringUtils = mockStatic(StringUtils.class);
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java:87]: Mocking the `StringUtils` seems to be little bit suspicious. There seems to be no mocks added as well? What is the requirement of mocking `StringUtils`?

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListenerTest.java [hunk 7]
```diff
@@ -49,22 +48,14 @@
...
-import static org.mockito.Matchers.any;
-import static org.mockito.Matchers.anyInt;
-import static org.mockito.Matchers.anyObject;
-import static org.mockito.Matchers.anyString;
-import static org.mockito.Matchers.eq;
-import static org.powermock.api.mockito.PowerMockito.mockStatic;
-import static org.powermock.api.mockito.PowerMockito.spy;
-import static org.powermock.api.mockito.PowerMockito.when;
-import static org.powermock.api.mockito.PowerMockito.whenNew;
+import static org.mockito.Mockito.*;
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListenerTest.java:51]: No * imports. Check other places as well.

**Review by madurangasiriwardena — APPROVED** [review]

**Review by ashensw — APPROVED** [review]

**Pull request merged**